go/vt/sqlparser/Makefile

```diff
@@ -15,7 +15,7 @@
 MAKEFLAGS = -s
 
 sql.go: sql.y
-	go run ./goyacc -fast-append -o sql.go sql.y
+	go run ./goyacc --fast-append --o sql.go -- sql.y
```

Comment:
you should not need the `--` separator here

Reply:
Yes, it works without it too. When exactly do we need it?

Reply:
`--` stops parsing regardless of what follows. we needed it to transition _onto_ `pflag`, because `pflag` allows positional and flag args to be mixed, whereas `flag` does not (first thing to not start with a `-` is considered the beginning of posargs, which broke anything with a subcommand tree). As an example:

```
./mycmd --foo "bar" subcommand --subflag "subval"
```

is parsed by `flag` into a single flag `--foo`, and `Args()` slice of: `[]string{"subcommand", "--subflag", "\"subval\""}`
but is parsed by `pflag` into flags of `--foo` and `--subflag` and an `Args()` slice of `[]string{"subcommand"}`

because our current subcommands are written on the assumption of `flag` parsing behavior, this would break because the top-level command's FlagSet did not define the `subflag` flag, so we inject the `--` to make it work in either case. since we're now 100% on `pflag`, we don't need it anymore

Reply:
That is a great explanation! Thank-you. Fixed.